service/controller/resource/monitoring/remotewriteconfig/resource.go

```diff
@@ -133,6 +135,12 @@ func (r *Resource) desiredConfigMap(cluster metav1.Object, name string, namespac
 func (r *Resource) getShardsCountForCluster(ctx context.Context, cluster metav1.Object, currentShardCount int) (int, error) {
 	headSeries, err := prometheusquerier.QueryTSDBHeadSeries(key.ClusterID(cluster))
 	if err != nil {
+		// If prometheus is not accessible (for instance, not running because this is a new cluster, we check if prometheus is accessible)
+		var dnsError *net.DNSError
```

Comment:
Coming from https://stackoverflow.com/questions/41669475/golang-idiomatic-way-to-detect-no-such-host-error